test-lib-openshift.sh

```diff
@@ -424,16 +444,39 @@ function ct_os_docker_login() {
 #                          In the format of name:tag ($image_name:latest by default)
 # Uses global REGISRTY_ADDRESS environment variable for arbitrary registry address.
 function ct_os_upload_image() {
+  local os_version="${1}" ; shift
   local input_name="${1}" ; shift
   local image_name=${input_name##*/}
   local imagestream=${1:-$image_name:latest}
   local output_name
+  local image_tagged="${image_name%:*}:${VERSION}"
 
-  output_name="${REGISRTY_ADDRESS:-172.30.1.1:5000}/$(oc project -q)/$imagestream"
+  if [ "${os_version}" != "v3" ] && [ "${os_version}" != "v4" ]; then
+    echo "You have to specify OpenShift version to upload an image."
+    echo "Either 'v3' or 'v4' is allowed"
+    return 1
+  fi
+
+  if [ "${os_version}" == "v3" ]; then
+    output_name="${REGISRTY_ADDRESS:-172.30.1.1:5000}/$(oc project -q)/$imagestream"
 
-  ct_os_docker_login
-  docker tag "${input_name}" "${output_name}"
-  docker push "${output_name}"
+    if ct_os_docker_login_v3; then
+      docker tag "${input_name}" "${output_name}"
+      docker push "${output_name}"
+      return 0
+    fi
+    return 1
+  fi
+  if [ "${os_version}" == "v4" ]; then
+    # Variable OCP4_ADDRESS is set in function ct_os_docker_login_v4
+    if ct_os_docker_login_v4; then
+      output_name="$OCP4_ADDRESS/$namespace/$image_tagged"
+      docker tag "${image_tagged}" "${output_name}"
+      docker push --tls-verify=false "${output_name}"
```

Comment:
These 3 commands are almost the same as in case of v3, so it's possible to have this piece of code only once and only define `$address`, `$namespace` and `$image_tagged` differently for v3 and v4 if needed. It's not a large piece of code though, so I don't have a big problem with keeping it as it is.

Reply:
Thanks for the review. I have addressed your comment. Please review it again.